starknet-rpc-test/get_transaction_receipt.rs

```diff
@@ -37,38 +36,39 @@ async fn get_transaction_receipt(
 
 #[rstest]
 #[tokio::test]
-async fn work_with_invoke_transaction(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
-    let madara = madara.await;
-    let rpc = madara.get_starknet_client();
+async fn work_with_invoke_transaction(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+    let rpc = madara.get_starknet_client().await;
 
-    let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
     let recepient = FieldElement::from_hex_be("0x123").unwrap();
     let transfer_amount = FieldElement::ONE;
-    let mut txs = madara
-        .create_block_with_txs(vec![Transaction::Execution(account.transfer_tokens(recepient, transfer_amount, None))])
-        .await?;
+
+    let mut txs = {
+        let mut madara_write_lock = madara.write().await;
+        let account = create_account(&rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
+        madara_write_lock
+            .create_block_with_txs(vec![Transaction::Execution(account.transfer_tokens(
+                recepient,
+                transfer_amount,
+                None,
+            ))])
+            .await?
+    };
 
     assert_eq!(txs.len(), 1);
     let rpc_response = match txs.remove(0).unwrap() {
         TransactionResult::Execution(rpc_response) => rpc_response,
         _ => panic!("expected execution result"),
     };
 
-    let invoke_tx_receipt = get_transaction_receipt(rpc, rpc_response.transaction_hash).await;
+    let invoke_tx_receipt = get_transaction_receipt(&rpc, rpc_response.transaction_hash).await;
     let fee_token_address = FieldElement::from_hex_be(FEE_TOKEN_ADDRESS).unwrap();
-    let expected_fee = FieldElement::from_hex_be("0x1d010").unwrap();
+    let expected_fee = FieldElement::from_hex_be("0xf154").unwrap();
```

Comment:
curious, why did the fees change?

Reply:
Those tests were not updated when https://github.com/keep-starknet-strange/madara/pull/1139 was merged.
Because their were not part of the CI
It is rectified with this PR.

Reply:
oh ok, makes sense